route/option.go

```diff
@@ -232,6 +236,103 @@ func ServiceGroups(serviceGroups []ServiceGroup) Option {
 	return serviceGroupsFunc(serviceGroups)
 }
 
+// Selector sets the given custom location selector. The location selector lets
+// you define a function which selects the location that will be inserted next
```

Comment:
```suggestion
// you define a function which selects the locations that will be inserted next
```

From ignorance I would expect that `model.Domain` holds multiple locations? If only one location will be inserted should it be an `int`? We should check to make sure 🙏🏻 .

Reply:
Yeah was thinking the same, it mimics the current API though. Maybe the reasoning was to also be able to return an empty domain? @larsbeck @nilsbeck @merschformann what do you think?